packages/compiler/src/core/helpers/type-name-utils.ts

```diff
@@ -182,7 +182,16 @@ function getInterfaceName(iface: Interface, options: TypeNameOptions | undefined
 }
 
 function getOperationName(op: Operation, options: TypeNameOptions | undefined) {
-  return `${getNamespacePrefix(op.namespace, options)}${getIdentifierName(op.name, options)}`;
+  let opName = getIdentifierName(op.name, options);
+  if (op.node.templateParameters.length > 0) {
+    // template
+    const params = op.node.templateParameters.map((t) => getIdentifierName(t.id.sv, options));
+    opName += `<${params.join(", ")}>`;
+  }
+  const prefix = op.interface
+    ? getInterfaceName(op.interface, options) + "."
+    : getNamespacePrefix(op.namespace, options);
+  return `${prefix}${opName}`;
 }
```

Comment:
@timotheeguerin , is there any case that template instantiated is true for operation's template arguments? I can only find it's true when the parent interface is used by alias, so skipped that part here for operation.

And about the template instantiated, is it only true when it's used by alias? I found for some statement like "op Op1 is Op2", the template instantiated is still false for Op2. is that expected?

Reply:
no operations are not always in an interface you can just do 

```
op test(): void;
```

Reply:
but yeah if it is in an interface `namespaceName.interfaceName.operationName` is correct

Reply:
looking at what you have done that looks good

Reply:
thanks for the confirmation. Then what if the interface is a template. should it still be namespaceName.interfaceName.operationName? or namespaceName.interfaceName<T>.operationName?

Reply:
yeah I would think so

Reply:
I think this looks perfect in the [playground](https://cadlplayground.z22.web.core.windows.net/prs/3183/)
![image](https://github.com/microsoft/typespec/assets/1031227/217ee42c-dda2-4a5c-8166-25ca6ba3afb5)